tower-http/src/compression/compression_filter.rs

```diff
@@ -0,0 +1,26 @@
+use crate::compression_utils::default_compression_filter_predicate;
+
+/// A filter which any response Parts needs to pass to be compressed
+pub trait CompressionFilter: Copy {
+    /// Predicate which takes response parts and returns true if the response should be compressed
+    fn should_compress(&self, parts: &http::response::Parts) -> bool;
+}
+
+impl<F> CompressionFilter for F where F: Fn(&http::response::Parts) -> bool + Copy {
+    fn should_compress(&self, parts: &http::response::Parts) -> bool {
+        (self)(parts)
+    }
+}
+
+/// Default compression filter that proxies `default_compression_filter_predicate` which looks at
+/// headers to determine whether compression is suitable
+#[derive(Default, Copy, Clone, Debug)]
+pub struct DefaultCompressionFilter {
+
+}
```

Comment:
This is equivalent to

```suggestion
#[derive(Default, Copy, Clone, Debug)]
#[non_exhaustive]
pub struct DefaultCompressionFilter;
```